src/plugins/data/public/ui/search_bar/create_search_bar.tsx

```diff
@@ -76,17 +94,21 @@ const defaultOnRefreshChange = (queryService: QueryStart) => {
 const defaultOnQuerySubmit = (
   props: StatefulSearchBarProps,
   queryService: QueryStart,
-  currentQuery: Query
+  currentQuery: Query,
+  uiActions: UiActionsStart,
+  abortControllerRef: React.MutableRefObject<AbortController | undefined>
 ) => {
   if (!props.useDefaultBehaviors) return props.onQuerySubmit;
-
   const { timefilter } = queryService.timefilter;
 
   return (payload: { dateRange: TimeRange; query?: Query }) => {
     const isUpdate =
       !_.isEqual(timefilter.getTime(), payload.dateRange) ||
       !_.isEqual(payload.query, currentQuery);
     if (isUpdate) {
+      // register ABORT_DATA_QUERY_TRIGGER
+      uiActions.addTriggerAction(ABORT_DATA_QUERY_TRIGGER, createAbortDataQueryAction());
+      abortControllerRef.current = new AbortController();
```

Comment:
What about moving to https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/search/search_service.ts ?